src/CoercibleUtils.hs

```diff
@@ -1,8 +1,58 @@
-module CoercibleUtils where
+{- |
+Utility functions for Coercible types.
+Primarily pulled from the package `newtype-generics`, and
+based on Conor McBride's Epigram work, but
+generalised to work over anything `Coercible`.
 
-import Data.Coerce
+>>> ala Sum foldMap [1,2,3,4]
+10
```

Comment:
What's  a bit disappointing about these examples is that they don't just work like that in ghci:

```
λ  ala Sum foldMap [1,2,3,4]

<interactive>:15:2: error:
    • Couldn't match representation of type ‘a0’ with that of ‘a'’
      ‘a'’ is a rigid type variable bound by
        the inferred type for ‘it’:
          forall a a'. (Num a, Coercible a a') => a'
        at <interactive>:15:2-26
    • In the ambiguity check for the inferred type for ‘it’
      To defer the ambiguity check to use sites, enable AllowAmbiguousTypes
      When checking the inferred type
        it :: forall a a'. (Num a, Coercible a a') => a'
```

Sometimes I think having explicit type annotations would be more honest. But maybe there's a more elegant solution?! Thoughts?

Reply:
I think we just need to give the list a type signature, unfortunately. In real code I don't think this will be that much of a problem, since coerce is usually used for type-safe performance gains, and that matters mostly in performant (generally non-ghci) code

Reply:
e.g. example must be

```haskell
ala Sum foldMap ([1,2,3,4] :: [Int]) :: Int
```

Reply:
just created a benchmark for that. it seems to run 2x slower than prelude's 'sum', which might have to do with Sum's monoid instance using 'coerce', and then 'getSum #.'

Reply:
Ok, I'll add it

Reply:
see PR #10